lib/room.rb

```diff
@@ -0,0 +1,10 @@
+class Room 
+  attr_accessor :number, :rate, :room_reservations, :blocked
+
+  def initialize(number, room_reservations:, blocked: false)
+    @number = number
+    @room_reservations = []
+    @blocked = blocked
```

Comment:
You have an instance variable here to keep track of the reservations for this room, but as far as I can tell you never use it.

Reply:
I'm not sure it makes sense to store `blocked` as a true or false value for the room. Whether or not a room is blocked depends on what date you're asking about, same as whether or not it's reserved. If you're keeping track of that here, the way to do it would be by storing a list of blocks that contain this room, and having a method `blocked(date_range)` that returns true or false.